src/Core/Kernel.php

```diff
@@ -49,29 +48,14 @@ private function prepareContainer()
         if (!$this->container) {
             $builder = new ContainerBuilder(new ParameterBag());
             $builder->addCompilerPass(new RegisterListenersPass());
-            $definition = new Definition(ContainerAwareEventDispatcher::class, [new Reference('service_container')]);
+            $definition = new Definition(EventDispatcher::class, [new Reference('service_container')]);
+            $definition->setPublic(true);
             $builder->setDefinition('event_dispatcher', $definition);
-
-            $this->loadServiceConfig($builder);
```

Comment:
So were the services not being used in that config file? I noticed you've gone and removed all of them.

Reply:
All of the removed code were basically dead ends; appears they were brought in on the initial building out of the Event Dispatcher in an abstract form. But they are not used at all in the current real Event Dispatcher form. Figured best to remove them this since it just was confusing as I tried to sort through this.